### PR TITLE
feat: scheduler (2/): add ClusterScore related structs, etc.

### PR DESCRIPTION
**Pull request by michaelawyu**
### Description of your changes

This PR is part of the PRs that implement the Fleet workload scheduling.

It features some structs for keeping scores in the scheduling process.

I have:

- [x] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested

- [x] Unit tests

### CODE REVIEW

File: pkg/scheduler/framework/score.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+type ClusterScore struct {
+	// TopologySpreadScore determines how much a binding would satisfy the topology spread
+	// constraints specified by the user.
+	TopologySpreadScore int
+	// AffinityScore determines how much a binding would satisfy the affinity terms
+	// specified by the user.
+	AffinityScore int
+	// PriorityScore determines how much a binding would satisfy the priority terms
+	// specified by the user.
+	PriorityScore int
```
Comment (ryanzhang-oss) [inline pkg/scheduler/framework/score.go:22]: we don't have that now, right?

Comment (michaelawyu) [reply pkg/scheduler/framework/score.go:22]: Yeah. Though the PriorityScore is mentioned in the PRD and the design doc, none of the P0 plugins touch this score :(

Comment (ryanzhang-oss) [reply pkg/scheduler/framework/score.go:22]: please remove them for this PR then

File: pkg/scheduler/framework/score.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+// Add adds a ClusterScore to another ClusterScore.
+func (s1 *ClusterScore) Add(s2 ClusterScore) {
```
Comment (zhiying-lin) [inline pkg/scheduler/framework/score.go:26]: just curious when we'll need this func

Comment (michaelawyu) [reply pkg/scheduler/framework/score.go:26]: We need them as multiple plugins will return their own scores and it is up to the scheduler to add them together.

Comment (zhiying-lin) [reply pkg/scheduler/framework/score.go:23]: oh, i see, thank for the info

**Review by zhiying-lin — APPROVED** [review]

**Pull request merged**